lib/server.js

```diff
@@ -30,7 +30,7 @@ class MockServer extends EventEmitter {
     this.onRead = typeof options.onRead === 'function' ? options.onRead : null;
 
     // Create a bson instance
-    this.bson = new BSON();
+    this.bson = BSON;
```

Comment:
If we're moving forward with a hard dependency on BSONv4, then I think the refactor here should be more complete. I think you can remove `this.bson` here, and just import `BSON` directly in `protocol.js`